cli/types/cypress.d.ts

```diff
@@ -3244,6 +3244,162 @@ declare namespace Cypress {
+     * @example  'Create React apps with no build configuration'
+     */
+    description: string
+
+    /**
+     * Minimum version supported. Should conform to Semantic Versioning as used in `package.json`.
+     * @see https://docs.npmjs.com/cli/v9/configuring-npm/package-json#dependencies
+     * @example '^=4.0.0 || ^=5.0.0'
+     * @example '^2.0.0'
+     */
+    minVersion: string
+  }
+
+  interface ResolvedComponentFrameworkDefinition {
+    /**
+     * A semantic, unique identifier. Must begin with `cypress-ct-` for third party implementations.
+     * @example: 'reactscripts', 'nextjs', 'cypress-ct-solid-js'
+     */
+    type: string
+
+    /**
+     * Used as the flag for `getPreset` for meta framworks, such as finding the webpack config for CRA, Angular, etc.
+     *
+     * configFramwork: () => {
+     *   return getSolidJsMetaFrameworkBundlerConfig()
+     * }
+     * It is also the name of the string added to `cypress.config`
+     *
+     * @example
+     *
+     * export default {
+     *   component: {
+     *     devServer: {
+     *       framework: 'create-react-app' // can be 'next', 'create-react-app', etc etc.
+     *     }
+     *   }
+     * }
+     */
+    configFramework: string
+
+    /**
+     * Library (React, Vue) or template (aka "meta framework") (CRA, Next.js, Angular)
+     */
+    category: 'library' | 'template'
+
+    /**
+     * Name displayed in Launchpad when doing initial setup.
+     * @example 'Solid.js', 'Create React App'
+     */
+    name: string
+
+    /**
+     * Supported bundlers.
+     */
+    supportedBundlers: Array<'webpack' | 'vite'>
+
+    /**
+     * Used to attempt to automatically select the correct framework/bundler from the dropdown.
+     * @example
+     *
+     * const SOLID_DETECTOR: Dependency = {
+     *   type: 'solid',
+     *   name: 'Solid.js',
+     *   package: 'solid-js',
+     *   installer: 'solid-js',
+     *   description: 'Solid is a declarative JavaScript library for creating user interfaces',
+     *   minVersion: '^1.0.0',
+     * }
+     */
+    detectors: CypressComponentDependency[]
+
+    /**
+     * Array of required dependencies. This could be the bundler and JavaScript library.
+     * It's the same type as `detectors`.
+     */
+    dependencies: (bundler: 'webpack' | 'vite', projectPath: string) => Promise<DependencyToInstall[]>
+
+    /**
+     * @internal
+     * This is used interally by Cypress for the "Create From Component" feature.
+     */
+    codeGenFramework?: 'react' | 'vue' | 'svelte' | 'angular'
+
+    /**
+     * @internal
+     * This is used interally by Cypress for the "Create From Component" feature.
+     * @example '*.{js,jsx,tsx}'
+     */
+    glob?: string
+
+    /**
+     * This is the path to get mount, eg `import { mount } from <mount_module>,
+     * @example: `cypress-ct-solidjs/src/mount`
+     */
+    mountModule: (projectPath: string) => Promise<string>
```

Comment:
Mentioned this in the tech-brief but I'm curious what you think of making the `ThirdPartyComponentFrameworkDefinition` have `mountModule: string` for simplicity?

Reply:
Good point, I was trying to make the APIs as similar as possible but it's probably not worth it - better to opt for some internal complexity and expose a nicer public API

Reply:
We don't need `mountModule` at all for third parties now... https://github.com/cypress-io/engineering-documentation/pull/71

Since we assume it's named export from the default entry point, we can assume `import { mount } from '<package>'` is valid.